### PR TITLE
PR: Modernise Development Process.

### PR DESCRIPTION
**Pull request by KelSolaar**
This PR modernises the development process and adopts some of the latest tools and approaches we have rolled over at [colour-science](https://github.com/colour-science), improves logging while also expanding the build process capabilities.

Code Quality
---

- Increased [pre-commit](https://pre-commit.com) usage.
- Replace *Yapf* with [Black](https://black.readthedocs.io) which is becoming the de-factor formatter for Python.
- Use [pyupgrade](https://github.com/asottile/pyupgrade) to automatically upgrade the code to a particular Python version, currently set to Python 3.8.
- Use [flynt](https://github.com/ikamensh/flynt) to ensure that strings are `f-Strings`, *pyupgrade* does it but more conservatively.
- Drop [nose](https://nose.readthedocs.io/en/latest/) as it is not maintained, fully replace with [pytest](https://docs.pytest.org/en/7.0.x/).

Docstrings Quality
---

Better [PEP 257](http://www.python.org/dev/peps/pep-0257/) compliance, the verification is performed with [PyDocstyle](https://github.com/PyCQA/pydocstyle). I have ignored a few checks that were deemed either irrelevant because with use Sphinx/reStructuredText and Black (and they produce conflicts) or required rewriting too much, e.g. the docstrings should have a one liner description:

- D104: Missing docstring in public package
  - Forces us to put docstrings in all the `__init__` files, we can reassess later.
- D200 : One-line docstring should fit on one line with quotes
  - Excessive amount of work to fix
- D202: No blank lines allowed after function docstring
  - Black puts a blank line.
- D205: 1 blank line required between summary line and description
- D301: Use r””” if any backslashes in a docstring
  - We escape all the backslashes. 
- D400: First line should end with a period
  - We use Sphinx/reStructuredText module headers.

Documentation
---

- New [PyData](http://pydata-sphinx-theme.readthedocs.io/) Sphinx theme.
- Re-organised the documentation.

Logging
---

- Many of the core definitions are logging to debug.
- Each module that uses logging defines a dedicated logger.

Build Process
---

- Artifacts are now being generated for the various configs & products the generators create: https://github.com/AcademySoftwareFoundation/OpenColorIO-Config-ACES/actions/runs/1974589422, https://github.com/AcademySoftwareFoundation/OpenColorIO-Config-ACES/actions/workflows/configuration-artifacts.yml

Next Step
---
- Typing Annotation  & Mypy

PS: Third time the charm! C.f. much pain in #38, #39.

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    git submodule update --init --recursive
+
+Poetry
+------
+
+The *OpenColorIO Configuration for ACES* repository adopts `Poetry <https://poetry.eustace.io>`__
+to help managing its dependencies, this is the recommended way to get started
+with development.
+
+Assuming `python>=3.7 <https://www.python.org/download/releases/>`__ is
```
Comment (doug-walker) [inline docs/installation.rst:28]: It says 3.8 on line 73.

File: docs/status.rst [hunk 2]
```diff
@@ -0,0 +1,38 @@
+..
+  SPDX-License-Identifier: CC-BY-4.0
+  Copyright Contributors to the OpenColorIO Project.
+
+Component Status
+================
+
++-------------------------------+----------------+----------------------------------------------------------------------------------+
+| Component                     | Status         | Notes                                                                            |
```
Comment (doug-walker) [inline docs/status.rst:9]: This table seems to be duplicated in the Readme?

**Review by doug-walker — DISMISSED** [review]

File: docs/status.rst (hunk 2 above)
Comment (KelSolaar) [reply docs/status.rst:9]: Yes it was on purpose but I can drop it!

File: docs/installation.rst (hunk 1 above)
Comment (KelSolaar) [reply docs/installation.rst]: Good catch thank you!

**Review by michdolan — APPROVED** [review]

**Pull request merged**